extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -433,4 +439,38 @@ void runtimeInitializedClasses(BuildProducer<RuntimeInitializedClassBuildItem> r
         runtimeInitializedClasses.produce(new RuntimeInitializedClassBuildItem(ObjectId.class.getName()));
         runtimeInitializedClasses.produce(new RuntimeInitializedClassBuildItem("com.mongodb.internal.dns.DefaultDnsResolver"));
     }
+
+    /**
+     * Ensure we have at most one customizer per Mongo client.
+     *
+     * @param beans the beans
+     * @param validation the producer used to report issues
+     */
+    @BuildStep
+    void validateMongoConfigCustomizers(BeanDiscoveryFinishedBuildItem beans,
+            BuildProducer<ValidationPhaseBuildItem.ValidationErrorBuildItem> validation) {
+        var type = DotName.createSimple(MongoClientCustomizer.class);
```

Comment:
We usually declare DotName as a constant, can you move it near to the other as constant at the top of the class?